doc/mgmt/SONiC_OC_Management_VRF_HLD.md

```diff
@@ -0,0 +1,301 @@
+Provide management framework capabilities to handle:
+- add/delete Management VRF
+- show Management VRF
+
+### 1.1.1 Functional Requirements
+Provide management framework support to existing SONiC capabilities with respect to Management VRF.
+Here is the link for the Management VRF HLD in SONiC, 
+https://github.com/Azure/SONiC/blob/310de5a9d649481e23ec3b048cbe90242c6e063f/mgmt-vrf-design-doc.md  
+
+### 1.1.2 Configuration and Management Requirements
+- CLI style configuration and show commands
+- REST API support
+- gNMI Support
+
+Details described in Section 3.
+
+Configurations not supported by this feature using management framework:
+
+- this feature does not provide the configuration option of binding Management VRF to a SNMP server. This is because the management framework does not support SNMP.
+
+- this feature does not provide the configuration option of using Management VRF for NTP. This is because SONiC NTP uses Management VRF automatically when configured, and uses default VRF if Management VRF is not configured.  
+It is up to the user to configure NTP server correctly. This feature does not have dependency on the NTP server configuration.
+
+- no support of configuring Management VRF for other IP services such as DNS, tftp, ftp etc.
+
+### 1.1.3 Scalability Requirements
+
+### 1.1.4 Warm Boot Requirements
+
+## 1.2 Design Overview
+
+### 1.2.1 Basic Approach
+Implement Management VRF support using transformer in sonic-mgmt-framework.
+
+### 1.2.2 Container
+All code changes will be done in management-framework container including:
+- XML file for the CLI
+- Python script to handle CLI request (actioner)
+- Jinja template to render CLI output (renderer)
+- OpenConfig YANG model for Management VRF
+- SONiC YANG model for Management VRF based on Redis DB schema of Management VRF 
+  (https://github.com/Azure/SONiC/wiki/Configuration#management-vrf)
+- transformer functions to 
+   * convert OpenConfig YANG model to SONiC YANG model for Management VRF configuration
+   * convert from SONiC YANG model to OpenConfig YANG model for Management VRF show
+
+### 1.2.3 SAI Overview
+
+# 2 Functionality
+
+## 2.1 Target Deployment Use Cases
+Manage/configure Management VRF via gNMI, REST and CLI interfaces
+
+## 2.2 Functional Description
+Provide CLI, gNMI and REST supports for Management VRF handling
+
+# 3 Design
+
+## 3.1 Overview
+
+Enhancing the management framework backend code and transformer methods to add support for Management VRF Handling
+
+## 3.2 DB Changes
+
+### 3.2.1 CONFIG DB
+This feature will allow the user to make/show Management VRF configuration changes to CONFIG DB
+
+### 3.2.2 APP DB
+
+### 3.2.3 STATE DB
+
+### 3.2.4 ASIC DB
+
+### 3.2.5 COUNTER DB
+
+## 3.3 Switch State Service Design
+
+### 3.3.1 Orchestration Agent
+
+### 3.3.2 Other Process
+
+## 3.4 SyncD
+
+## 3.5 SAI
+
+## 3.6 User Interface
+
+### 3.6.1 Data Models
+YANG models needed for Management VRF handling in the management framework:
```

Comment:
Will one data model be locked for create if Mgmt VRF is already created using another model? Can user use both models at the same time.

Reply:
Only openconfig yang model is used for user configuration. The sonic yang model is used internally to translate the openconfig yang to redis DB schema for management VRF. Hope this is what you are asking.

Thx.